tools/flang1/flang1exe/exterf.c

```diff
@@ -2097,18 +2097,40 @@ export_dt(int dtype)
       }
     } else /* 'null' descriptor */
       lzprintf(outlz, " %d", 0);
+
+    /* If align pragma value is smaller than its original alignment,
+     * then align pragma should have no effect
+     */
+    if (DTA(dtype) == alignment(dtype))
+      lzprintf(outlz, " %d", (int)DTA(dtype));
+    else
+      lzprintf(outlz, " %d", 0);
+
     break;
   case TY_UNION:
   case TY_STRUCT:
   case TY_DERIVED:
     /*  print dtype and  descriptor entry */
-    lzprintf(outlz, " %d %d %d %d", (int)DTY(dtype + 1), (int)DTY(dtype + 2),
-             (int)DTY(dtype + 3), (int)DTY(dtype + 4));
+    lzprintf(outlz, " %d %d %d", (int)DTY(dtype + 1), (int)DTY(dtype + 2),
+             (int)DTY(dtype + 3));
+    if (DTA(dtype) > DTY(dtype + 4))
+      lzprintf(outlz, " %d", (int)DTA(dtype));
+    else
+      lzprintf(outlz, " %d", (int)DTY(dtype + 4));
     break;
 
   case TY_CHAR:
   case TY_NCHAR:
     lzprintf(outlz, " %d", (int)DTY(dtype + 1));
+
+    /* If align pragma value is smaller than its original alignment,
+     * then align pragma should have no effect
+     */
+    if (DTA(dtype) == alignment(dtype))
```

Comment:
Same here.

Reply:
> Same here.

Yes, you are correct. I have corrected the code using >.